presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -908,6 +911,11 @@ public FragmentProperties setDistribution(PartitioningHandle distribution, Metad
                 return this;
             }
 
+            if (isPreferLocalUnion(session) && distribution.equals(SOURCE_DISTRIBUTION)) {
```

Comment:
This is added to support the case 
`bucketed table with 32 bucket UNION all bucketed table with 16 bucket`, the PartitioningHandle is set to table with 16 buckets